### PR TITLE
fix: field expansion - fill only relevant fields while traversing abstract types

### LINKED ISSUE #112: Field Expansion for Union and Interfaces misbehaving
**Description**
When Union or  interfaces are sharing the same  types `too` as in the bellow schema, `too` is expanded for all types(Foo and Bar),  We expect shared fields are expand only for fields used in query

Given Schema
``` GQL

        type Query {
          uBaz: Baz
        }
        union Baz = Foo | Bar
        type Foo {
          foo: String
          too: String
        }
        type Bar {
          bar: Int
          too: String
        }
      `
```
and Query
```GQL
 `
            query {
              uBaz {
                ... on Foo {
                  foo
                  too
                }
                ... on Bar {
                  bar
                }
              }
            }
          `
```

Expected `info.fieldExpansion`  is

```
                  Object {
                    "Bar": Object {
                      "bar": Object {
                        Symbol(LeafFieldSymbol): true,
                      },
                    },
                    "Foo": Object {
                      "foo": Object {
                        Symbol(LeafFieldSymbol): true,
                      },
                      "too": Object {
                        Symbol(LeafFieldSymbol): true,
                      },
                    },
                  }
```

But Actual  `info.fieldExpansion`  is
```
                 Object {
                    "Bar": Object {
                      "bar": Object {
                        Symbol(LeafFieldSymbol): true,
                      },
                      "too": Object {     <--------- `too` is not expected as its not in the query   
                        Symbol(LeafFieldSymbol): true,
                      },
                    },
                    "Foo": Object {
                      "foo": Object {
                        Symbol(LeafFieldSymbol): true,
                      },
                      "too": Object {
                        Symbol(LeafFieldSymbol): true,
                      },
                    },
                  }
           
```

### PR DESCRIPTION
**Pull request by boopathi**
fix #112

### CODE REVIEW

**Comment by boopathi** [comment]
👍

File: src/resolve-info.ts [hunk 1]
```diff
@@ -172,12 +174,24 @@ type FragmentsType = GraphQLResolveInfo["fragments"];
 type GraphQLNamedOutputType = GraphQLNamedType & GraphQLOutputType;
 type GraphQLObjectLike = GraphQLInterfaceType | GraphQLObjectType;
 
-const memoizedGetReturnType = memoize2(getReturnType);
-const memoizedHasField = memoize2(hasField);
-const memoizedResolveEndType = memoize(resolveEndType);
-const memoizedGetPossibleTypes = memoize2(getPossibleTypes);
-const memoizedExpandFieldNodeType = memoize4(expandFieldNodeType);
-const memoizedExpandFieldNode = memoize4(expandFieldNode);
+const MEMOIZATION = true;
```
Comment (ponimas) [inline src/resolve-info.ts:177]: Could you explain why we need that constant?

Comment (boopathi) [reply src/resolve-info.ts:177]: Only for debugging purposes when changing / updating the implementation of field filtering.

**Comment by topicus** [comment]
👍

**Pull request merged**